client/mainwindow.cpp

```diff
@@ -9,37 +9,39 @@
 
 MainWindow::MainWindow(QWidget* parent)
   : QMainWindow(parent)
-  , ui(std::make_unique<Ui::MainWindow>())
 {
-    ui->setupUi(this);
-    ui->lineEdit->setPlaceholderText("Enter what you want send to server here...");
-    ui->sendButton->setDisabled(true);
-    ui->disconnectButton->setDisabled(true);
-    ui->lineEdit->setDisabled(true);
-    textBrowser = QPointer<QTextBrowser>(ui->textBrowser);
-    connection = std::make_shared<Connection>(textBrowser);
-    connect(ui->connectButton, &QPushButton::clicked, this, &MainWindow::onConnectClicked);
-    connect(ui->disconnectButton, &QPushButton::clicked, this, &MainWindow::onDisconnectClicked);
-    connect(ui->sendButton, &QPushButton::clicked, this, &MainWindow::onSendClicked);
+    connection_tab = std::make_unique<Connection_tab>();
+    QVBoxLayout* processes_layout = new QVBoxLayout();
+    processes_tab = new QWidget();
+
+    processes_layout->addWidget(new QLabel("Nothing special here yet."));
+    processes_tab->setLayout(processes_layout);
+    connection_tab->get_main_widget()->addTab(processes_tab, "Processes");
+
+    setCentralWidget(connection_tab->get_main_widget());
+    setWindowTitle("Client");
+    resize(400, 300);
+    connection_tab->disable_communication_ui();
+
+    connect(connection_tab.get(), &Connection_tab::connectionButtonClicked, this, &MainWindow::onConnectClicked);
+    connect(connection_tab.get(), &Connection_tab::disconnectionButtonClicked, this, &MainWindow::onDisconnectClicked);
+    connect(connection_tab.get(), &Connection_tab::sendButtonClicked, this, &MainWindow::onSendClicked);
 }
 
 void MainWindow::onConnectClicked()
 {
-    if (!connection->is_connected())
+    if (!connection.is_connected())
     {
         try
         {
-            connection->connect();
-            ui->connectionIndicator->setChecked(true);
-            ui->sendButton->setEnabled(true);
-            ui->disconnectButton->setEnabled(true);
-            ui->connectButton->setDisabled(true);
-            ui->lineEdit->setEnabled(true);
+            connection = Connection(QPointer<QTextBrowser>(connection_tab->get_text_browser()));
+            connection.connect();
+            connection_tab->enable_communication_ui();
```

Comment:
This call should be a slot connected to signal emitted in `Connection` object when connection was established successfully.

On other hand when connection will become broken or close it could emit proper signals for it and communication UI could be disabled automatically.